### PR TITLE
Improve realm constructor logic

### LINKED ISSUE #566: Improve logging messages for android
We should give a checklist of things to check when getting the `Missing Realm Constructor` message. See #375

### LINKED ISSUE #583: Visual Studio Code debugging
I have an issue with Realm and VSCode debugging, i'm unable to debug because apparently the Realm constructor is missing.

```
[vscode-react-native] Finished executing: react-native run-ios --simulator iPhone 6s --project-path ios
node --debug-brk=35498 launchReactNative.js 9090 
Debugger listening on [::]:35498
[vscode-react-native] Downloaded debuggerWorker.js (Logic to run the React Native app) from the Packager.
[vscode-react-native] Starting debugger app worker.
[vscode-react-native] Established a connection with the Proxy (Packager) to the React Native application
[vscode-react-native] Debugging session started successfully.
Missing Realm constructor - please ensure RealmReact framework is included!
Module AppRegistry is not a registered callable module.
```

Then, if I `Stop Remote JS Debugging`, it works.

However, **everything works fine** in remote js debugging or not when I use `run-android` or `run-ios` instead of VSCode.

So, I guess it's not an integration issue and something went wrong between Realm and VSCode debugging.
#### Environment

osx@10.11.6
node@6.2.2
npm@3.9.5
react-native@0.32.0
realm@0.14.3
vscode-react-native@0.1.6
vscode@1.4.0

Related to #374, #562 et #571.

cc @alazier @appden

### LINKED ISSUE #648: Update docs/error message with instructions for fixing the missing constructor issue
Users continue to hit #375 

We should update the error message with steps to try to address the issue as outlined here: https://github.com/realm/realm-js/issues/375#issuecomment-259596699

### LINKED ISSUE #736: VS code debugging crashes with react native and realm-js, also debug crashes on nuclide
Actual Behavior

debugging crashes on react native 0.39, vscode 1.7.2, realm 0.15.0
Similar crashing happens on nuclide too.
Expected Behavior

Could able to debug the app with Realm-JS
Software versions

React-Native VS Code extension version: 1.7.2
VSCode version: 1.7.2
OS platform and version: mac os 10.12.1
NodeJS version: 7.2.1
React Native version: 0.39
Outputs (Include if relevant)

[vscode-react-native] Starting debugger app worker.
[vscode-react-native] Established a connection with the Proxy (Packager) to the React Native application
[vscode-react-native] Debugging session started successfully.
Error: socket hang up
at doRequest (/Users/admin/Documents/Development/react-native/lang-app/languageApp/node_modules/sync-request/index.js:37:11)
at makeRequest (/Users/admin/Documents/Development/react-native/lang-app/languageApp/.vscode/.react/index.ios.bundle:73531:14)
at sendRequest (/Users/admin/Documents/Development/react-native/lang-app/languageApp/.vscode/.react/index.ios.bundle:73561:14)
at Object.createSession (/Users/admin/Documents/Development/react-native/lang-app/languageApp/.vscode/.react/index.ios.bundle:73452:1020)
at /Users/admin/Documents/Development/react-native/lang-app/languageApp/.vscode/.react/index.ios.bundle:73080:31
at loadModuleImplementation (/Users/admin/Documents/Development/react-native/lang-app/languageApp/.vscode/.react/index.ios.bundle:188:1)
at guardedLoadModule (/Users/admin/Documents/Development/react-native/lang-app/languageApp/.vscode/.react/index.ios.bundle:140:8)
at _require (/Users/admin/Documents/Development/react-native/lang-app/languageApp/.vscode/.react/index.ios.bundle:124:1)
at /Users/admin/Documents/Development/react-native/lang-app/languageApp/.vscode/.react/index.ios.bundle:72899:18
at loadModuleImplementation (/Users/admin/Documents/Development/react-native/lang-app/languageApp/.vscode/.react/index.ios.bundle:188:1)
Realm failed to connect to the embedded debug server inside the app. If attempting to use Chrome debugging from a device, ensure the device is reachable on the same network as this machine.

### PR DESCRIPTION
**Pull request by kristiandupont**
This fixes #648, fixes #583, fixes #736, fixes #566

### CODE REVIEW

**Comment by kristiandupont** [comment]
I've added #752 for that. I think it would be nice to release this to possibly move things forward for people building RN on Windows.

File: lib/index.js [hunk 1]
```diff
@@ -18,43 +18,72 @@
 
 'use strict';
 
-function node_require(module) {
+console.log("Up to date");
```
Comment (sorenvind) [inline lib/index.js]: ?

File: lib/index.js [hunk 2]
```diff
@@ -18,43 +18,72 @@
...
-function isNode() {
-    return typeof process == 'object' && (('' + process) == '[object process]' || typeof jest == 'object')
-}
 
-// function isElectronRenderer() {
-//     return isNode() && process.type === 'renderer'
-// }
+function getContext() {
+    if (typeof process !== 'undefined') {
+        if(typeof process === 'object' && process + '' === '[object process]') {
```
Comment (sorenvind) [inline lib/index.js]: What about collapsing this to simply the inner condition? The outer one doesn't give any additional value.

File: lib/index.js (hunk 1 above)
Comment (kristiandupont) [inline lib/index.js]: Whoops, that was for debugging..

**Comment by deodeveloper** [comment]
I don't think it fixes the debug issue in VS code #736

**Pull request merged**